src/engine/mode_manager.cpp

```diff
@@ -262,12 +262,20 @@ GameMode *ModeEngine::GetTop()
 // Returns a pointer to a game mode on the stack
 GameMode *ModeEngine::Get(uint32 index)
 {
-    if(_game_stack.size() < index)
-        return nullptr;
-    else
-        return _game_stack.at(_game_stack.size() - index);
+    GameMode* result = nullptr;
+
+    assert(_game_stack.size() > index);
+    if (_game_stack.size() > index) {
+        result = _game_stack.at(_game_stack.size() - index - 1);
+    }
+
+    return result;
 }
 
+size_t ModeEngine::Size() const
+{
+    return _game_stack.size();
```

Comment:
I'll remove the Size function.

The change above it is technically a fix.  There was an off-by-one array bug before.  However, I do not think that function is technically used either.